source/aks/patching-artifactory.html.md.erb

```diff
@@ -0,0 +1,159 @@
+
+```command
+kubectl get pods -n artifactory 
+```
+
+![](Images/artifactory_pic02.png)
+
+This command lists running pods.
+
+```command
+kubectl describe pods -n artifactory artifactory-oss-0 
+```
+
+![](Images/artifactory_pic03.png)
+
+This command provides detailed information about the desired pods, confirming the version (Image: 7.71.5)
+
+4. Version Checking:
+
+• Visit the following website to look for latest self hosted version:[Artifactory Release Information] (https://jfrog.com/help/r/jfrog-release-information/artifactory-7.77.5-self-hosted)
+
+5. Pre-update Testing:
+
+• Before updating the PTL box, conduct testing to ensure no issues
+
+• Download the repository from [GitHub] (https://github.com/hmcts/cnp-flux-config)
+
+• File location: apps/artifactory/artifactory/…
+
+• Create a new file named "artifactory-sbox.yaml" with the provided code snippet
+
+```command
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: artifactory-oss
+  namespace: artifactory
+spec:
+  serviceName: "artifactory"
+  replicas: 2
+  selector:
+    matchLabels:
+      app: artifactory-oss
+  template:
+    metadata:
+      labels:
+        app: artifactory-oss
+    spec:
+      containers:
+        - image: docker.bintray.io/jfrog/artifactory-oss:7.77.3
+          name: artifactory-oss
+```
+
+[Example PR] (https://github.com/hmcts/cnp-flux-config/pull/29609/files)
+
+6. Run Test:
+
+• Add the below code to kustomization.yaml file for the test to run:
+
+```command
+patchesStrategicMerge:-../../artifactory/artifactory-sbox.yaml
+```
+
+7. Push Changes:
+
+• Raise a PR and push changes out for review and approval
+
+8. Update Version:
+
+• For the PTLSBOX, remove "artifactory-sbox.yaml" file and its corresponding code from kustomization.yaml
+
+• Update the version in "artifactory.yaml" to 7.77.3
+
+```command
+      containers:
+        - image: docker.bintray.io/jfrog/artifactory-oss:7.77.3
+          name: artifactory-oss
+          env:
+```
+
+[Example PR] (https://github.com/hmcts/cnp-flux-config/pull/29665/files)
+
+9. Push Changes:
+
+• Raise a PR and push changes out for review and approval
+
+• If you can go to the [Artifactory](https://artifactory.sandbox.platform.hmcts.net/ui/repos/tree/General/) site after the upgrade and checking the packages and repositories to ensure they are working.
+
+10. Release Draft and Version Bump:
+
+• Publish the draft release in "chart-neuvector"
+
+• Bump the version in "flux-config" from 1.5.4 to 1.5.5
+
+11. Push Changes:
+
+• Raise a PR and push changes out for review and approval
+
+Troubleshooting Steps:
+
+• To resolve namespace issues in ["chart-neuvector"] (https://hmcts.github.io/ops-runbooks/aks/updating-neuvector-policies.html)
+
+	• Delete specific files from the azure file share
+
+	• Files to delete: "admission_control.backup" and "crd.backup"
+
+	• Delete the HR on the cluster and let it redeploy
+
+	• Confirm proper functioning after redeployment
+
```

Comment:
For https://github.com/hmcts/ops-runbooks/pull/225/files#diff-902428882fa500d61f112f609ba06a87a0c7f657956d63f57ddec843f57a7802R145-R153 just say that we had specific issues with neuvector which should be resolved and the details can be found in the linked page. We don't want to give the impression that this needs done every time